
package MyPacket;
option optimize_for = LITE_RUNTIME;

enum MessageType 
{
    PKT_CS_LOGIN = 1;
	PKT_SC_LOGIN = 2;
	
	PKT_CS_CHAT = 3;
	PKT_SC_CHAT = 4;
	
	PKT_CS_MOVE = 5;
	PKT_SC_MOVE = 6;

	PKT_SC_FEED = 7;

	PKT_CS_SERVER_STATUS = 8;
	PKT_SC_SERVER_STATUS = 9;

	PKT_CS_ORDER = 10;
}

message LoginRequest
{
    required int32 PlayerId = 1;
}

 message Position 
 {
    required float x = 1;
    required float y = 2;
	required float z = 3;
 }
	
message LoginResult
{
    required int32 PlayerId = 1;
	required string PlayerName = 2;
	required Position PlayerPos = 3;
}

message ChatRequest
{
    required int32 PlayerId = 1;
    required string PlayerMessage = 2;
}

message ChatResult
{
    required string PlayerName = 1;
    required string PlayerMessage = 2;
}

message MoveRequest
{
    required int32 PlayerId = 1;
	required Position PlayerPos = 2;
}

message MoveResult
{
    required int32 PlayerId = 1;
	required Position PlayerPos = 2;
}

message Depth
{
	required int32 Count = 1;
	required int32 Qty = 2;
	required float Price = 3;
}

message Feed
{
	required string ProductCode = 1;
	repeated Depth BidDepth = 2;
	repeated Depth AskDepth = 3;
}

message ServerStatus
{
	required int32 SessionCount = 1;
}

message Order
{
	required int32 OrderNumber = 1;
	required int32 OrderQty = 2;
	required float OrderPrice = 3;
	required string ProductCode = 4;
}